parts/k8s/kubernetescustomscript.sh

```diff
@@ -61,557 +63,77 @@ else
-    sed -i "s|<kubernetesClusterAutoscalerVmType>|$(echo $VM_TYPE | base64)|g" "/etc/kubernetes/addons/cluster-autoscaler-deployment.yaml"
-    sed -i "s|<kubernetesClusterAutoscalerVMSSName>|$(echo $PRIMARY_SCALE_SET)|g" "/etc/kubernetes/addons/cluster-autoscaler-deployment.yaml"
-}
-
-configACIConnectorAddon() {
-    ACI_CONNECTOR_CREDENTIALS=$(printf "{\"clientId\": \"$(echo $SERVICE_PRINCIPAL_CLIENT_ID)\", \"clientSecret\": \"$(echo $SERVICE_PRINCIPAL_CLIENT_SECRET)\", \"tenantId\": \"$(echo $TENANT_ID)\", \"subscriptionId\": \"$(echo $SUBSCRIPTION_ID)\", \"activeDirectoryEndpointUrl\": \"https://login.microsoftonline.com\",\"resourceManagerEndpointUrl\": \"https://management.azure.com/\", \"activeDirectoryGraphResourceId\": \"https://graph.windows.net/\", \"sqlManagementEndpointUrl\": \"https://management.core.windows.net:8443/\", \"galleryEndpointUrl\": \"https://gallery.azure.com/\", \"managementEndpointUrl\": \"https://management.core.windows.net/\"}" | base64 -w 0)
-
-    openssl req -newkey rsa:4096 -new -nodes -x509 -days 3650 -keyout /etc/kubernetes/certs/aci-connector-key.pem -out /etc/kubernetes/certs/aci-connector-cert.pem -subj "/C=US/ST=CA/L=virtualkubelet/O=virtualkubelet/OU=virtualkubelet/CN=virtualkubelet"
-    ACI_CONNECTOR_KEY=$(base64 /etc/kubernetes/certs/aci-connector-key.pem -w0)
-    ACI_CONNECTOR_CERT=$(base64 /etc/kubernetes/certs/aci-connector-cert.pem -w0)
-
-    sed -i "s|<kubernetesACIConnectorCredentials>|$ACI_CONNECTOR_CREDENTIALS|g" "/etc/kubernetes/addons/aci-connector-deployment.yaml"
-    sed -i "s|<kubernetesACIConnectorResourceGroup>|$(echo $RESOURCE_GROUP)|g" "/etc/kubernetes/addons/aci-connector-deployment.yaml"
-    sed -i "s|<kubernetesACIConnectorCert>|$(echo $ACI_CONNECTOR_CERT)|g" "/etc/kubernetes/addons/aci-connector-deployment.yaml"
-    sed -i "s|<kubernetesACIConnectorKey>|$(echo $ACI_CONNECTOR_KEY)|g" "/etc/kubernetes/addons/aci-connector-deployment.yaml"
-}
-
-configAddons() {
-    if [[ "${CLUSTER_AUTOSCALER_ADDON}" = True ]]; then
-        configClusterAutoscalerAddon
-    fi
-
-    if [[ "${ACI_CONNECTOR_ADDON}" = True ]]; then
-        configACIConnectorAddon
+function holdWALinuxAgent() {
+    if [[ $OS == $UBUNTU_OS_NAME ]]; then
+        # make sure walinuxagent doesn't get updated in the middle of running this script
+        retrycmd_if_failure 20 5 30 apt-mark hold walinuxagent || exit $ERR_HOLD_WALINUXAGENT
     fi
 }
 
 testOutboundConnection
+holdWALinuxAgent
 waitForCloudInit
 
-if [[ ! -z "${MASTER_NODE}" ]]; then
-    echo "executing master node provision operations"
+if $FULL_INSTALL_REQUIRED; then
     installEtcd
-else
-    echo "skipping master node provision operations, this is an agent node"
+    installDeps
+    installContainerRuntime
+    installNetworkPlugin
+    installContainerd
+    extractHyperkube
+    installFlexVolDrivers
+else 
+    echo "Golden image; skipping dependencies installation"
```

Comment:
For now skipping all installs if using golden image for simplicity but we should discuss checking each dependency individually.

Reply:
I discussed this with Cecile - I think having some sort of check would be good for partial upgrades. Use cases:

1. in-place security updates
2. During test automation, we need to be able to test new versions of Kubernetes, Azure-CNI, moby, and so on. We shouldn't need to recreate the VHD in those scenarios